packages/elements/src/components/ino-datepicker/ino-datepicker.tsx

```diff
@@ -272,71 +283,59 @@ export class Datepicker implements ComponentInterface {
     this.valueChange.emit(formattedDate);
   }
 
-  @Listen('clickEl')
-  inoIconClickedHandler() {
-    this.focusInputField();
-    this.flatpickr.toggle();
-  }
-
-  focusInputField = () => {
-    const currentFocus: Element = document.activeElement;
-    const input = this.el.querySelector('input') as HTMLInputElement;
-
-    // Don't change focus if current focus is an input field (e.g. time picker)
-    if (currentFocus.tagName !== 'input') {
-      input.focus();
-    }
-  };
-
-  private static INPUT_CLASSES = [
-    'cur-year',
-    'flatpickr-hour',
-    'flatpickr-minute',
-    'flatpickr-time',
-  ];
-
-  private elementIsInput(element: Element) {
-    const elementClasses = element.className;
-    return Datepicker.INPUT_CLASSES.some((cl) => elementClasses.includes(cl));
-  }
-
   /**
    * Emits when the value of the datepicker changes.
    * The value can be found in `event.detail`
    */
   @Event() valueChange!: EventEmitter<string>;
 
   componentDidLoad() {
-    if (!this.disabled) this.create();
+    if (!this.disabled) {
+      this.create();
+    }
   }
 
   disconnectedCallback() {
     this.dispose();
   }
 
-  private static NUMBERS_WITH_SPECIAL_CHARS = /(\d[^a-z]+)/g;
-
   private create() {
     this.dispose();
 
-    if (this.disabled) return;
+    if (this.disabled) {
+      return;
+    }
 
     const sharedOptions: Partial<BaseOptions> = {
-      allowInput: true,
+      allowInput: !this.inoRange,
       clickOpens: false,
       ignoredFocusElements: [],
       locale: getDatepickerLocale(this.el),
       onValueUpdate: (_, newValue) => this.valueChange.emit(newValue),
-      onChange: () => this.focusInputField(),
     };
 
-    const typeSpecificOptions: Partial<BaseOptions> = this.getTypeSpecificOptions();
+    const typeSpecificOptions: Partial<BaseOptions> = getTypeSpecificOption(
+      this.inoType,
+      {
```

Comment:
Why not passing the Element itself? The Props should be public vars so you don'T have to map all props

Reply:
Which element do you mean? `this`? I have to remap them sooner or later because they are `ino`-prefixed

Reply:
Ah missed that points, Makes sense 👍